ur_robot_driver/include/ur_robot_driver/comm/stream.h

```diff
@@ -107,7 +108,12 @@ class URStream : public TCPSocket
 protected:
   virtual bool open(int socket_fd, struct sockaddr* address, size_t address_len)
   {
-    return ::connect(socket_fd, address, address_len) == 0;
+    if (::connect(socket_fd, address, address_len) == -1)
```

Comment:
Should we check `!= 0` just in case?

Reply:
We could, but [according to the docs](https://linux.die.net/man/3/connect) `connect()` should only return `0` or `-1`.